### PR TITLE
TF-2565: Promote the app

### PR DESCRIPTION
**Pull request by nqhhdev**
### Ticket:

- #2565 

### Resolved

- Android 

https://github.com/linagora/tmail-flutter/assets/99852347/db99a266-b298-4d0d-ba47-5b467397a0b6

- Web Desktop

https://github.com/linagora/tmail-flutter/assets/99852347/1568180a-34fd-41b5-8153-8963d399a5a6

### CODE REVIEW

**Comment by chibenwa** [comment]
Really cool.

File: web/worker_service/style.css [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	display: flex; 
+	justify-content: space-between; 
+	align-items: center; 
+	padding: 10px 20px; 
+	color: #fff; 
+} 
+
+.sheet-content { 
+	padding: 20px; 
+}
```
Comment (hieutbui) [inline web/worker_service/style.css:24]: indent

File: web/worker_service/worker_service.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  }
+  scriptLoaded = true;
+  var scriptTag = document.createElement('script');
+  scriptTag.src = 'main.dart.js';
+  scriptTag.type = 'application/javascript';
+  document.body.append(scriptTag);
+}
+
+function fetchServiceWorker() {
+if ('serviceWorker' in navigator) {
```
Comment (hieutbui) [inline web/worker_service/worker_service.js:21]: indent

File: web/worker_service/worker_service.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  // If service worker doesn't succeed in a reasonable amount of time,
+  // fallback to plaint <script> tag.
+  setTimeout(() => {
+    if (!scriptLoaded) {
+      console.warn(
+        '[TwakeMail] fetchServiceWorker(): Failed to load app from service worker. Falling back to plain <script> tag.',
+      );
+      loadMainDartJs();
+    }
+  }, 4000);
```
Comment (hieutbui) [inline web/worker_service/worker_service.js:70]: Should add const variable for this

**Review by hieutbui — APPROVED** [review]

**Comment by dab246** [comment]
How to open the app once it has been installed?

File: web/index.html [hunk 4]
```diff
@@ -55,78 +57,44 @@
             <div class="loader"></div>
         </div>
     </div>
+
+    <div class="bottom-sheet">
+      <div class="sheet-header">
+        <span class="open_on_mobile_title">Open Twake Mail in...</span>
```
Comment (dab246) [inline web/index.html:63]: How to support multiple languages?

**Comment by hoangdat** [comment]
> How to open the app once it has been installed?

Because the deep link take time to register, so we will postpone `opening` app to the next release. This fix almost for the people who did not install app

File: web/worker_service/worker_service.js [hunk 5]
```diff
@@ -0,0 +1,139 @@
+const androidStore = 'https://play.google.com/store/apps/details?id=com.linagora.android.teammail';
+const iosStore = 'https://apps.apple.com/app/twake-mail/id1587086189';
+const iosPlatform = 'iOS';
+const androidPlatform = 'android';
+const otherPlatform = 'other';
+var timeoutDuration = 4000;
```
Comment (dab246) [inline web/worker_service/worker_service.js:6]: Should be `const`

File: web/index.html (hunk 4 above)
Comment (nqhhdev) [reply web/index.html:63]: I'll find solution for this case

File: web/index.html [hunk 6]
```diff
@@ -32,10 +32,13 @@
   <title>Twake Mail</title>
   <link rel="manifest" href="manifest.json">
   <link rel="stylesheet" type="text/css" href="splash/style.css">
+  <link rel="stylesheet" type="text/css" href="worker_service/style.css">
   <meta content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no" name="viewport">
 
   <script type="application/javascript" src="assets/packages/flutter_inappwebview/assets/web/web_support.js" defer></script>
   <script src="splash/splash.js"></script>
+  <script src="worker_service/worker_service.js"></script>
+   <script src="i18n/translater.js"></script>
```
Comment (hieutbui) [inline web/index.html:41]: indent

File: web/worker_service/worker_service.js [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+  document.body.style.overflow = "hidden";
+  smartBanner.style.top = "16px";
+}
+
+function closeSmartBanner() {
+  console.info('[TwakeMail] closeSmartBanner(): Closing the smart banner.');
+  const smartBanner = document.querySelector(".smart-banner");
+  smartBanner.style.display = "none";
+  document.body.style.overflow = "auto";
+  smartBanner.style.bottom = "-100%";
```
Comment (hieutbui) [inline web/worker_service/worker_service.js:138]: Banner open base on top attribute. So we should set `bottomSheet.style.top` back to 0 here

File: web/i18n/translater.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
+const i18n = {};
+
+function getUserLanguage() {
+  console.info(`[Twake Mail] Current Language: `, navigator.language);
+  return navigator.language.split('-')[0];
```
Comment (hieutbui) [inline web/i18n/translater.js:5]: Need check for case when `navigator.language` is null || 'undefine' || null so return default language

**Review by hieutbui — APPROVED** [review]

**Review by dab246 — APPROVED** [review]
lgtm

**Pull request merged**